src/libsystemd-network/dhcp6-option.c

```diff
@@ -167,6 +167,33 @@ int dhcp6_option_append_fqdn(uint8_t **buf, size_t *buflen, const char *fqdn) {
         return r;
 }
 
+int dhcp6_option_append_user_class(uint8_t **buf, size_t *buflen, char **user_class) {
+        _cleanup_free_ uint8_t *p = NULL;
+        size_t total = 0, offset = 0;
+        char **s;
+
+        assert_return(buf && *buf && buflen && user_class, -EINVAL);
+
+        STRV_FOREACH(s, user_class) {
+                size_t len = strlen(*s);
+
+                if (len > 65535)
```

Comment:
doesn't matter, but i think writing this as 0xffff is a bit easier on the eyes. Or maybe UINT16_MAX would be even better